api/src/main/java/me/lokka30/treasury/api/economy/currency/CurrencyManager.java

```diff
@@ -0,0 +1,248 @@
+     *
+     * @param currencyId currency id you want the currency of
+     * @return currency if present, empty optional otherwise
+     * @author MrIvanPlays
+     * @since v1.0.0
+     */
+    public Optional<Currency> getCurrency(@NotNull UUID currencyId) {
+        Objects.requireNonNull(currencyId, "currencyId");
+        return Optional.ofNullable(BY_ID.get(currencyId));
+    }
+
+    /**
+     * Returns the {@link Currency} bound to the specified {@link String} {@code currencyName}, if present.
+     *
+     * @param currencyName the name of the currency needed
+     * @return currency if present, empty optional otherwise
+     * @author MrIvanPlays
+     * @since v1.0.0
+     */
+    public Optional<Currency> getCurrency(@NotNull String currencyName) {
+        if (!hasCurrency(currencyName)) {
+            return Optional.empty();
+        }
+        UUID currencyId = null;
+        for (String key : BY_NAME.keySet()) {
+            if (key.equalsIgnoreCase(currencyName)) {
+                currencyId = BY_NAME.get(key);
+                break;
+            }
+        }
+        if (currencyId != null) {
+            return getCurrency(currencyId);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Returns the {@link Currency} bound to the specified {@link Character} {@code currencyCharacter}, if present.
+     *
+     * @param currencyCharacter the character of the currency needed
+     * @return currency if present, empty optional otherwise
+     * @author MrIvanPlays
+     * @since v1.0.0
+     */
+    public Optional<Currency> getCurrency(@NotNull Character currencyCharacter) {
+        if (!hasCurrency(currencyCharacter)) {
+            return Optional.empty();
+        }
+        return Optional.of(BY_ID.get(BY_CHAR.get(currencyCharacter)));
+    }
+
+    /**
+     * Converts the specified {@code value} from the {@link Currency} {@code from} to the  {@link Currency} {@code to} by
+     * using their {@link Currency#getConversionCoefficient()}.
+     *
+     * @param from  the currency to convert from
+     * @param to    the currency to convert to
+     * @param value the value to convert
+     * @return converted value
+     * @throws IllegalArgumentException if one or both of the currencies are not registered
+     * @author MrIvanPlays, MrNemo64
+     * @since v1.0.0
+     */
+    public double convertCurrency(@NotNull Currency from, @NotNull Currency to, double value) {
+        Objects.requireNonNull(from, "from");
+        Objects.requireNonNull(to, "to");
+        if (!hasCurrency(from.getCurrencyId()) || !hasCurrency(to.getCurrencyId())) {
+            throw new IllegalArgumentException("Specified currencies are unknown to the currency manager");
+        }
+        if (from.getConversionCoefficient() == 0 || to.getConversionCoefficient() == 0 || value == 0) {
+            return 0;
+        }
+        // thanks MrNemo64 for the formula
```

Comment:
Pun on a comment or something https://github.com/lokka30/Treasury/issues/49#issuecomment-986255592 so if anyone more doesn't understand the formula he can just read the explanation

Reply:
done.